common.py

```diff
@@ -524,3 +529,71 @@ def build_sriov_network_operator_check_permissions() -> bool:
     rsh = host.LocalHost()
     ret = rsh.run("podman pull registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16")
     return ret.success()
+
+
+def git_repo_setup(repo_dir: str, *, repo_wipe: bool = True, url: str, branch: str = "master") -> None:
+    exists = os.path.exists(repo_dir)
+    if exists and not repo_wipe:
+        return
+
+    if exists:
+        shutil.rmtree(repo_dir)
+
+    logger.info(f"Cloning repo {url} to {repo_dir}")
+    Repo.clone_from(url, repo_dir, branch=branch)
+
+
+@dataclasses.dataclass
+class ContainerInfo:
+    name: str
+    envvar: str
+    containerfile: str
+    registry: str
+    project: str
+    full_tag: str = dataclasses.field(init=False)
+
+    def __post_init__(self) -> None:
+        self.full_tag = f"{self.registry}/{self.project}/{self.name}:latest"
+
+
+def start_image_registry(rsh: host.Host, client: K8sClient) -> str:
+    reglocal_dir_name, reglocal_hostname, reglocal_listen_port, reglocal_id = reglocal.ensure_running(rsh)
+
+    reglocal.ocp_trust(client, reglocal_dir_name, reglocal_hostname, reglocal_listen_port)
+
+    registry = f"{reglocal_hostname}:{reglocal_listen_port}"
+
+    return registry
```

Comment:
This should move to `reglocal`.

`reglocal` already provides you with `ensure_running()` and `ocp_trust()`. As there is a use for a helper method that ties those two togeter, that also belongs there...